### PR TITLE
feat: Filter templates by workspace

### PR DESCRIPTION
**Pull request by gt2345**
## Ticket

[MD-378]

## Description

Modify the get templates endpoint to accept workspace id.
This is used in displaying templates at workspace details page.

## Test Plan

`GET /templates` endpoint should accept workspace id and return the filtered result.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[MD-378]: https://hpe-aiatscale.atlassian.net/browse/MD-378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/templates/api.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ func (a *TemplateAPIServer) GetTemplates(
 	if req.Name != "" {
 		q.Where("name ILIKE  ('%%' || ? || '%%')", req.Name)
 	}
+	if req.WorkspaceId != 0 {
```
Comment (stoksc) [inline master/internal/templates/api.go:51]: remind me, is 0 the uncategorized / default namespace, and should we allow filtering by it?

File: e2e_tests/tests/template/test_template.py [hunk 2]
```diff
@@ -306,3 +306,36 @@ def test_rbac_template_exp_create() -> None:
             )
             assert proc.returncode == 1
             assert "not found" in proc.stderr
+
+
+@pytest.mark.e2e_cpu
+def test_get_templates_by_workspace() -> None:
```
Comment (stoksc) [inline e2e_tests/tests/template/test_template.py:312]: can you exchange this test for a test like what is in `api_intg_test.go`, where all the other template api tests are.

File: master/internal/templates/api.go (hunk 1 above)
Comment (gt2345) [reply master/internal/templates/api.go:51]: Default workspace is 1

File: e2e_tests/tests/template/test_template.py (hunk 2 above)
Comment (gt2345) [reply e2e_tests/tests/template/test_template.py]: Of course

**Review by stoksc — APPROVED** [review]

**Pull request merged**